neqo-transport/src/packet.rs

```diff
@@ -854,39 +914,57 @@ mod tests {
         assert!(encoder.is_empty());
     }
 
-    const SAMPLE_RETRY: &[u8] = &[
+    const SAMPLE_RETRY_27: &[u8] = &[
         0xff, 0xff, 0x00, 0x00, 0x1b, 0x00, 0x08, 0xf0, 0x67, 0xa5, 0x50, 0x2a, 0x42, 0x62, 0xb5,
         0x74, 0x6f, 0x6b, 0x65, 0x6e, 0xa5, 0x23, 0xcb, 0x5b, 0xa5, 0x24, 0x69, 0x5f, 0x65, 0x69,
         0xf2, 0x93, 0xa1, 0x35, 0x9d, 0x8e,
     ];
+
+    const SAMPLE_RETRY_28: &[u8] = &[
+        0xff, 0xff, 0x00, 0x00, 0x1c, 0x00, 0x08, 0xf0, 0x67, 0xa5, 0x50, 0x2a, 0x42, 0x62, 0xb5,
+        0x74, 0x6f, 0x6b, 0x65, 0x6e, 0xf7, 0x1a, 0x5f, 0x12, 0xaf, 0xe3, 0xec, 0xf8, 0x00, 0x1a,
+        0x92, 0x0e, 0x6f, 0xdf, 0x1d, 0x63,
+    ];
+
     const RETRY_TOKEN: &[u8] = b"token";
 
+    fn pick_retry_version() -> (&'static [u8], QuicVersion) {
+        if random(1)[0] % 2 == 0 {
+            (SAMPLE_RETRY_27, QuicVersion::Draft27)
+        } else {
+            (SAMPLE_RETRY_28, QuicVersion::Draft28)
+        }
+    }
```

Comment:
Let's not randomize this.  The reason we have the loop on the `build_retry_multiple` function is to exercise the version.  I would factor out the content of build_retry_single into a function that takes a quic_version parameter, then have:

```rust
fn build_retry_single(v: QuicVersion, sample: &[u8]) {
...
}
#[test]
fn build_retry_27() {
  build_retry_single(QuicVersion::Draft27, SAMPLE_RETRY_27);
}
#[test]
fn build_retry_28() {
  build_retry_single(QuicVersion::Draft28, SAMPLE_RETRY_28);
}
#[test]
fn build_retry_multiple() {
for i in 0..32 {
  build_retry_27();
  build_retry_28();
}
}
```

Same for `invalid_retry`

Reply:
given time constraints I'm going to do this work in a separate PR.